src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1481,22 +1483,23 @@ namespace winrt::TerminalApp::implementation
                                                                     const PasteFromClipboardEventArgs eventArgs)
     {
         co_await winrt::resume_foreground(Dispatcher(), CoreDispatcherPriority::High);
-
-        TerminalPage::PasteFromClipboard(eventArgs);
+        
+        TerminalPage::PasteFromClipboard(eventArgs, _settings->GlobalSettings().RemoveTabsOnPaste());
```

Comment:
Watch out - since you're accessing a member of `this` after a `co_await`, it's possible that `this` won't be valid anymore. You'll need to do the whole 
```c++
auto weakThis = ...;
co_await ...;
if (auto strongThis{ weakThis })
{
    // do the thing with `this` here
}
```
dance that's common throughout this file

Reply:
```diff
- if (auto strongThis{ weakThis })
+ if (auto strongThis{ weakThis.get() })
```

Reply:
Changed this, but we did have some issues with your other suggestion (explained below)